### DISCUSSION TITLE
Looking for MOVE.EXE with LFN support

### DISCUSSION THREAD

**Discussion by severach** [dosemu2/dosemu2#1926] [General]
I need a command line DOS MOVE.EXE similar to Microsoft MSDOS that can rename and move files with Long File Names.

FreeDOS move.exe: no LFN support even in FD13.

Odi's LREN.EXE: Always says "File Not Found". Can only rename, can't move.

Win 98 & Me MOVE.EXE: LFN support but converts file names to all upper case. This is not an emulator bug because VC renames with the correct case. Maybe someone has a patch that removes the upper case conversion.

4Dos 8.00 builtin move: No LFN support

Edit : Referenced in #1933

**Comment by stsp** [comment]
If you think some of these tools
behave inadequately, then please
open the tickets with the proper
description of the problem.
Moved to chat.

**Comment by stuaxo** [comment]
I think 4Dos's "ren" can do LFNs if you have enabled LFNs in 4DOS (I think you may have to set an env var or pass an option on the commandline?)

**Comment by severach** [comment]
4DOS fully supports LFN with REN, MOVE, and other commands but is only enabled by default when it detects a Win95 compatible environment. DOSEMU is not detected as Win95. Enable it with:

`4dos //Win95LFN=Yes`

or
```
C:\4DOS.INI
Win95LFN=Yes
```

or with `SHELLHIGH`

**Reply by andrewbird** [reply]
Do you know the mechanism for the detection, it seems like something that dosemu should support?

**Reply by skitt** [reply]
The overall Windows 95 detection in 4DOS involves calling 2Fh/1600h to determine whether Windows is running in enhanced mode, then checking that the DOS major version is 7 or greater but less than 20 (OS/2), and finally calling 21h/71A0h on the boot drive and checking whether it succeeds.

Once Windows 95 is detected, LFN support is enabled/disabled on each volume depending on the result of 21h/71A0h.

**Reply by andrewbird** [reply]
> involves calling 2Fh/1600h to determine whether Windows is running in enhances mode

Ohh, that seems like a show stopper. Thanks!
